radixdlt-core/radixdlt/src/main/java/com/radixdlt/universe/UniverseModule.java

```diff
@@ -43,15 +48,22 @@ private int magic(Universe universe) {
 
 	@Provides
 	@Singleton
-	private Universe universe(RuntimeProperties properties, Serialization serialization) {
-		try {
-			byte[] bytes = Bytes.fromBase64String(properties.get("universe"));
-			Universe u = serialization.fromDson(bytes, Universe.class);
-			UniverseValidator.validate(u, Sha256Hasher.withDefaultSerialization());
-			return u;
-		} catch (DeserializeException e) {
-			throw new IllegalStateException("Error while deserialising universe", e);
+	private Universe universe(RuntimeProperties properties, Serialization serialization) throws IOException {
+		var universeString = properties.get("universe");
+		final Universe u;
+		// TODO: better way of managing universes
+		if (Strings.isNotBlank(universeString)) {
+			byte[] bytes = Bytes.fromBase64String(universeString);
+			u = serialization.fromDson(bytes, Universe.class);
+		} else {
+			var universeFileName = properties.get("universe.location", ".//universe.json");
+			var file = new File(universeFileName);
+			try (FileInputStream universeInput = new FileInputStream(file)) {
+				u = serialization.fromJson(IOUtils.toString(universeInput), Universe.class);
+			}
 		}
+		UniverseValidator.validate(u, Sha256Hasher.withDefaultSerialization());
+		return u;
 	}
 
```

Comment:
Perhaps it might be more readable if code would be refactored as follows:

```
	private Universe universe(RuntimeProperties properties, Serialization serialization) throws IOException {
		var universeString = properties.get("universe");
		var universe = Strings.isNotBlank(universeString)
				? loadFromString(universeString, serialization)
				: loadFromFile(properties, serialization);

		UniverseValidator.validate(universe, Sha256Hasher.withDefaultSerialization());
		return universe;
	}

	private Universe loadFromString(String universeString, Serialization serialization) throws DeserializeException {
		var bytes = Bytes.fromBase64String(universeString);

		return serialization.fromDson(bytes, Universe.class);
	}

	private Universe loadFromFile(RuntimeProperties properties, Serialization serialization) throws IOException {
		var universeFileName = properties.get("universe.location", ".//universe.json");

		try (var universeInput = new FileInputStream(new File(universeFileName))) {
			return serialization.fromJson(IOUtils.toString(universeInput), Universe.class);
		}
	}
```